src/transformers/integrations/integration_utils.py

```diff
@@ -251,6 +253,14 @@ def _objective(trial: optuna.Trial, checkpoint_dir=None):
             if getattr(trainer, "objective", None) is None:
                 metrics = trainer.evaluate()
                 trainer.objective = trainer.compute_objective(metrics)
+
+            # Free GPU memory
+            del trainer.model_wrapped
+            del trainer.model
+            gc.collect()
+            trainer.accelerator.clear()
+            torch.cuda.empty_cache()
+
```

Comment:
Instead of doing this you can just call : 

```python
from accelerate.utils.memory import release_memory
trainer.model_wrapped, trainer.model = release_memory(trainer.model_wrapped, trainer.model)
trainer.accelerator.clear()
```

Reply:
Done!